Providers/StaminaProviderConfig.jsx

```diff
@@ -0,0 +1,119 @@
+// /* eslint-disable react/destructuring-assignment */
+import React, { Component, createContext } from "react";
+import { AsyncStorage } from "react-native";
+import axios from "react-native-axios";
+
+const { Provider, Consumer } = createContext();
+
+class StaminaProvider extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      stamina: null,
+      date: null
+    };
+  }
+
+  async componentDidMount() {
+    const storedResult = await this.loadFromStorage();
+    try {
+      const resp = await axios.get("http://worldtimeapi.org/api/ip");
+
+      const date = new Date(resp.data.datetime);
+      const storedDate = new Date(storedResult);
+
+      if (this.state.date) {
+        if ((date - storedDate) / 1000 > 86400) {
```

Comment:
this probably also should be abstracted. Maybe avoid numbers like 86400, use a constant or use other functions to compare dates more verbosely